### DISCUSSION TITLE
Sensor needs an automation to update?

### DISCUSSION THREAD

**Discussion by marcokreeft87** [delzear/hass-formulaoneapi#40] [Q&A]
Am I right that the sensor needs to be updated by calling a service? It does not update by itself

**Comment by delzear** [comment]
Nope, It auto updates every 10 minutes.

**Reply by marcokreeft87** [reply]
Hmm weird. It doesnt do that on my HA. State is always on 'Scheduled'. Even yesterday and the day before that

![image](https://user-images.githubusercontent.com/10223677/193589757-d38c507f-be1e-404e-a483-375379b5bb62.png)

**Reply by delzear** [reply]
Maybe but it updates the attributes. This was not ment to be a public project, so "It works" was fine for me! I do not have time to work on it more the a few hours a year, so these things do not get solved! Sorry for that!

**Reply by marcokreeft87** [reply]
I get that. No worries :) I might fork this and make it work for me than, is that ok?

**Reply by delzear** [reply]
I would invite you to merge here your changes. The api behind this sensor is located here my offices. If they go down, I try to post here that they are down!
Also, you could build something amazing and I don't want to miss out on that!

**Reply by marcokreeft87** [reply]
> I would invite you to merge here your changes. The api behind this sensor is located here my offices. If they go down, I try to post here that they are down! Also, you could build something amazing and I don't want to miss out on that!

Good point! I will do that :)

**Reply by marcokreeft87** [reply]
I've created a pull request. This will fix the sensor not updating. I will look at dividing the code into seperate files for readability and hope to add a live event feature when I have time

**Reply by marcokreeft87** [reply]
Is it an option for you to put all the ergast endpoints behind your cache?

**Reply by delzear** [reply]
I tried to implement has much has I could in the short time I had. I don't remember which one I have implemented.
I used the same calls has Ergast did so you can try. If you're missing one, just tell me I will try to implement it.

**Reply by marcokreeft87** [reply]
I'm missing a lot tbf. For example current/last/results. I will continu the development with the ergast url since its easy to switch. When you implemented the urls I will switch back to your api :)

**Reply by marcokreeft87** [reply]
But I dont think the ergast api will go down since there are so many application depending on it

**Reply by delzear** [reply]
Since you are in the development stage, I don't think there will be any problem but we cannot go live because he will block us like he did a year ago!

**Reply by marcokreeft87** [reply]
Ah ok, didnt know that. Well if you could implement the current/last/results endpoint for now than I got a pull request for you

**Reply by delzear** [reply]
It's online now!

**Reply by marcokreeft87** [reply]
Nice! I will try to make time now

**Reply by marcokreeft87** [reply]
Whats the url? It isnt https://ergastcache.mobiusmedia.ca/f1/current/last/results.json right?

**Reply by delzear** [reply]
yes

**Reply by marcokreeft87** [reply]
Well thats not working

**Reply by delzear** [reply]
Should work now! There is more then one result!!! Made a type result needed an "s". You URL should work!

**Comment by marcokreeft87** [comment]
Made a PR. Should we try to get to get into the HACS default library?

**Reply by delzear** [reply]
Yes sure! I have no idea how!

**Comment by marcokreeft87** [comment]
Check this out : https://github.com/marcokreeft87/formulaone-card.
Lets get both into HACS :)

Still need to fix a couple of things, after that I will try to get both into HACS. Frontend is a lot easier tho

**Reply by delzear** [reply]
Love it! Let's go!
